src/collection.js

```diff
@@ -600,6 +612,16 @@ export default class Collection {
       lastModified: options.lastModified,
       headers: options.headers
     })
+      // Check if server is empty (flushed or changed).
+      .then(changes => {
+        const firstSync =           (!this.lastModified);
+        const lastModifiedUpdated = (changes.lastModified !== this.lastModified);
+        const emptyCollection =     (changes.changes.length === 0);
+        syncResultObject.serverEmpty = (!firstSync &&
+                                        lastModifiedUpdated &&
+                                        emptyCollection);
```

Comment:
Nit: Also, how about adding a `markFlushed` method to `SyncResultObject`? This would allow eventually performing more checks/updates at the result object level. (unsure about that)